# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/rene/valgrind/linux-2.6.9.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -dur valgrind-2.2.0/coregrind/vg_unsafe.h valgrind-2.2.0-p/coregrind/vg_unsafe.h
--- valgrind-2.2.0/coregrind/vg_unsafe.h	2004-11-02 22:03:17.000000000 +0100
+++ valgrind-2.2.0-p/coregrind/vg_unsafe.h	2004-11-02 20:49:06.000000000 +0100
@@ -47,7 +47,17 @@
 #include <sys/socket.h>   /* for struct msghdr */
 #include <linux/sockios.h>/* for SIOCOUTQ */
 #include <sys/un.h>       /* for sockaddr_un */
+
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
+
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,9)
 #include <net/if.h>       /* for struct ifreq et al */
+#endif
 #include <net/if_arp.h>   /* for struct arpreq */
 #include <net/route.h>    /* for struct rtentry */
 #ifdef HAVE_LINUX_COMPILER_H
